frontend-web/src/coffee-call/coffee-call.tsx

```diff
@@ -35,11 +33,24 @@ export const CoffeeCall: Component = () => {
           <button onClick={() => protocol.start(username)}>Start</button>
         </Show>
       </div>
+
+      <Show when={coffeeCall().state === CoffeeCallState.announced}>Coffee call announced!</Show>
+      <Show when={coffeeCall().state === CoffeeCallState.inProgress}>Coffee call in progress!</Show>
+      <Show when={coffeeCall().state === CoffeeCallState.canceled}>Coffee call canceled!</Show>
+
+      <Show when={coffeeCall().state !== CoffeeCallState.inactive && coffeeCall().state !== CoffeeCallState.canceled}>
+        On board: {participantsAsList()}
+      </Show>
       <ul>
-        <For each={allMessages()}>{(message) =>
-          <li>{message.name} {message.type}s a coffee call</li>
+        <For each={coffeeCall().messages}>{(message) =>
+          <li>{message.name} {message.type}s a coffee call @{message.broadcastAt.toLocaleString()}</li>
         }</For>
-      </ul>
+      </ul >
     </>
   );
 };
+
+function fromWithDefault<T>(source: Parameters<typeof from<T>>[0], initial: T): Accessor<T> {
+  const signal = from(source);
+  return createMemo(() => signal() ?? initial);
+}
```

Comment:
That's awesome! Should be right in SolidJS itself.

As it is so generic and "lib-like", a component is probably not the best place to put it into.